dhcp-controller/controllers/dhcpserver_controller.go

```diff
@@ -173,114 +203,272 @@ func (r *DHCPServerReconciler) ensureServer(request reconcile.Request,
+			Namespace: v.Namespace,
+		},
+		Spec: kubevirtv1.VirtualMachineSpec{
+			Running: &t,
+			Template: &kubevirtv1.VirtualMachineInstanceTemplateSpec{
+				Spec: kubevirtv1.VirtualMachineInstanceSpec{
+					AccessCredentials: []kubevirtv1.AccessCredential{{
+						UserPassword: &kubevirtv1.UserPasswordAccessCredential{
+							Source: kubevirtv1.UserPasswordAccessCredentialSource{
+								Secret: &kubevirtv1.AccessCredentialSecretSource{
+									SecretName: v.Name + "-pwd",
+								},
+							},
+							PropagationMethod: kubevirtv1.UserPasswordAccessCredentialPropagationMethod{
+								QemuGuestAgent: &kubevirtv1.QemuGuestAgentUserPasswordAccessCredentialPropagation{},
+							},
+						},
+					}},
+					Domain: kubevirtv1.DomainSpec{
+						Memory: hugepageMemory,
+						Devices: kubevirtv1.Devices{
+							Disks: []kubevirtv1.Disk{{
+								Name: "containerdisk",
+								DiskDevice: kubevirtv1.DiskDevice{
+									Disk: &kubevirtv1.DiskTarget{
+										Bus: "virtio",
+									},
+								},
+							}, {
+								Name: "cloudinitdisk",
+								DiskDevice: kubevirtv1.DiskDevice{
+									Disk: &kubevirtv1.DiskTarget{
+										Bus: "virtio",
+									},
+								},
+							}, {
+								Name: "kc-disk",
+							}, {
+								Name: v.Name + "-disk",
+							}},
+							Interfaces: interfaceList,
+						},
+						Resources: kubevirtv1.ResourceRequirements{
 							Limits: corev1.ResourceList{
 								"memory": *memLimit,
+								"cpu":    *cores,
 							},
 							Requests: corev1.ResourceList{
 								"memory": *memReq,
+								"cpu":    *cores,
 							},
 						},
-						ImagePullPolicy: corev1.PullIfNotPresent,
-						Name:            v.Name,
-						Env: []corev1.EnvVar{
-							{
-								Name:  "BIND_INTERFACE_IP",
-								Value: interfaceIps,
-							},
-						},
-						VolumeMounts: []corev1.VolumeMount{{
-							Name:      "dnsmasq-cfg",
-							MountPath: "/etc/dnsmasq.d/",
-						}},
-					}},
-					ServiceAccountName: "dhcpserver-controller-manager",
-					Volumes: []corev1.Volume{{
-						Name: "dnsmasq-cfg",
-						VolumeSource: corev1.VolumeSource{
-							ConfigMap: &corev1.ConfigMapVolumeSource{
+					},
+					Networks: networkList,
+					Volumes: []kubevirtv1.Volume{{
+						Name: v.Name + "-disk",
+						VolumeSource: kubevirtv1.VolumeSource{
+							ConfigMap: &kubevirtv1.ConfigMapVolumeSource{
 								LocalObjectReference: corev1.LocalObjectReference{
 									Name: v.Name,
 								},
-								Items: []corev1.KeyToPath{{
-									Key:  "dnsmasq.conf",
-									Path: "dnsmasq.conf",
-								}},
+							},
+						},
+					}, {
+						Name: "kc-disk",
+						VolumeSource: kubevirtv1.VolumeSource{
+							Secret: &kubevirtv1.SecretVolumeSource{
+								SecretName: "dhcp-controller-kc",
+							},
+						},
+					}, {
+						Name: "containerdisk",
+						VolumeSource: kubevirtv1.VolumeSource{
+							ContainerDisk: &kubevirtv1.ContainerDiskSource{
+								Image: "docker.io/platform9/dhcpserver:v1",
```

Comment:
Lets add a constant for this image name.

Reply:
Done